### ISSUE TITLE
Failures on not-yet-published blog posts' meta tags

### ISSUE THREAD

**Issue by grahamc** [lycheeverse/lychee#1594]
Our CI pipeline builds our static site and runs lychee on it. Most of our pages have a canonical rel , and an og:url, among others, that point to the final production domain:

```
<link rel="canonical" href="https://determinate.systems/posts/xxxx">
<meta property="og:url" content="https://determinate.systems/posts/xxxx">
```

and to solve this, we've taken the unfortunate step of blanket-excluding our own domain from our checks. This seems... sub-optimal, and I think I'd like it better if I could explicitly exclude tags like these?

**Comment by mre** [comment]
Yup. Although I wonder how such an option might look like. `--exclude-tags rel,property` would not be enough, as you're targeting `rel="canonical"` and `property="og:url"` specifically.
Open for suggestions on the syntax here.

The alternative would be to count these links as verbatim URLs and exclude them by default, unless `--include-verbatim` is set. Alternatively, we could have our own flag like `--exclude-meta-urls` or so, but I'm not sure if it warrants a separate CLI flag. It's quite specific to HTML as well.

**Comment by dan-lee** [comment]
What if we could address tags similar to [CSS attribute selectors](https://developer.mozilla.org/en-US/docs/Web/CSS/Attribute_selectors)?
E.g. `--exclude-tags 'link[rel="canonical"]' --exclude-tags 'meta[property^="og:"]'`.

Not sure if this goes too far though, just wanted to give a suggestion because we are facing the same issue.

**Comment by mre** [comment]
I like that. In order to support that, we'd have to add a CSS attribute parser to our HTML extractors (html5gum and html5ever). There is [servo/selectors](https://docs.rs/selectors/latest/selectors/), which apparently provides `match` functions for CSS attributes.

**Comment by katrinafyi** [comment]
I would like to ask, is the problem that the URLs are not-yet-published pages which would be created from the local HTML files you're checking?

If so, an alternative solution would be to remap pages from your domain to the local files. This would let lychee check that the links correctly exist in the to-be-uploaded files. 

If that makes sense :)

**Comment by dan-lee** [comment]
Yes, that makes sense and works well :) Thanks for the workaround!

**Comment by katrinafyi** [comment]
I'm going to close this for the same reason as #1918 :) Let me know if you want to keep it open.

The workaround is now documented at https://lychee.cli.rs/recipes/local-folder/. Hopefully, this is very clear, otherwise do let me know. 

Further down the line, we might also try to turn the workaround into proper CLI arguments to make it easier to use.
